src/vcpkg/vcpkgpaths.cpp

```diff
@@ -672,9 +679,39 @@ namespace vcpkg
                                                        m_pimpl->m_config_dir,
                                                        *this);
 
+            auto make_relative_to_manifest = [&](std::string overlay_path) -> std::string {
+                auto const p = Path(overlay_path);
+                if (p.is_relative())
+                {
+                    Path ret = m_pimpl->m_config.directory / overlay_path;
+                    return ret.c_str();
+                }
+                return overlay_path;
+            };
```

Comment:
```suggestion
            auto resolve_relative_to_config = [&](const std::string& overlay_path) {
                return (m_pimpl->m_config.directory / overlay_path).native();
            };
```

1. This isn't a "make" function; that tends to indicate that it's going to change the value in-place rather than calculate a new one.
2. The function was named "make relative", but what it's actually doing is making a path absolute.
3. It doesn't need to copy the input string, so it can accept by `const&`.
4. `operator/` already has the "if the second parameter is absolute it overwrites" behavior:

https://github.com/microsoft/vcpkg-tool/blob/9a92af89d17755c2a449f416fde5b578ca3e8329/src/vcpkg/base/files.cpp#L1008-L1022